src/packet/ipv4.rs

```diff
@@ -39,29 +41,67 @@ pub struct Ipv4 {
     payload: Vec<u8>,
 }
 
-/// Calculates the checksum of an IPv4 packet
-pub fn checksum(packet: &Ipv4Packet) -> u16be {
-    use packet::Packet;
+/// Represents an IPv4 pseudo header
+///
+/// 0          8          16          24         32
+/// +---------------------------------------------+
+/// |                  source_ip                  |
+/// +----------------------+----------------------+
+/// |                destination_ip               |
+/// +-----------+----------+----------------------+
+/// |   zeros   | protocol | tcp/udp length       |
+/// +-----------+---------------------------------+
+#[packet]
+pub struct Ipv4PseudoHeader {
+    #[construct_with(u8, u8, u8, u8)]
+    source: Ipv4Addr,
+    #[construct_with(u8, u8, u8, u8)]
+    destination: Ipv4Addr,
+    zeros: u8,
```

Comment:
See comments in issue #146 here - https://github.com/libpnet/libpnet/pull/146/files#r47442790.

Relatedly, it's really confusing for me having multiple pull requests open for the same functionality - I feel like I'm ending up reviewing the same code multiple times, and things are going to end up getting lost, such as this. Is there any way this could all be consolidated into a single place? Perhaps you and @little-dude could set up a shared repo which you both have access to, and make pull requests from there.